src/Drupal/Commands/core/MigrateRunnerCommands.php

```diff
@@ -0,0 +1,706 @@
+            // Rollback in reverse order.
+            $migrations = array_reverse($migrations);
+            foreach ($migrations as $migrationId => $migration) {
+                $this->initProgressBar($migration, $options);
+                $executable = new MigrateExecutable($migration, $this->getMigrateMessage(), $executableOptions);
+                // drush_op() provides --simulate support.
+                drush_op([$executable, 'rollback']);
+            }
+        }
+    }
+
+    /**
+     * Stop an active migration operation.
+     *
+     * @command migrate:stop
+     *
+     * @drupal-dependencies migrate
+     *
+     * @param string $migrationId
+     *   The ID of migration to stop.
+     *
+     * @aliases mst,migrate-stop
+     *
+     * @topics docs:migrate
+     *
+     * @validate-module-enabled migrate
+     */
+    public function stop(string $migrationId): void
+    {
+        /** @var \Drupal\migrate\Plugin\MigrationInterface $migration */
+        $migration = $this->getMigrationPluginManager()->createInstance($migrationId);
+        switch ($migration->getStatus()) {
+            case MigrationInterface::STATUS_IDLE:
+                $this->logger()->warning(dt('Migration @id is idle', ['@id' => $migrationId]));
+                break;
+            case MigrationInterface::STATUS_DISABLED:
+                $this->logger()->warning(dt('Migration @id is disabled', ['@id' => $migrationId]));
+                break;
+            case MigrationInterface::STATUS_STOPPING:
+                $this->logger()->warning(dt('Migration @id is already stopping', ['@id' => $migrationId]));
+                break;
+            default:
+                $migration->interruptMigration(MigrationInterface::RESULT_STOPPED);
+                $this->logger()->log('status', dt('Migration @id requested to stop', ['@id' => $migrationId]));
+                break;
+        }
+    }
+
+    /**
+     * Reset an active migration's status to idle.
+     *
+     * @command migrate:reset-status
+     *
+     * @drupal-dependencies migrate
+     *
+     * @param string $migrationId
+     *   The ID of migration to reset.
+     *
+     * @aliases mrs,migrate-reset-status
+     *
+     * @topics docs:migrate
+     *
+     * @validate-module-enabled migrate
+     */
+    public function resetStatus(string $migrationId): void
+    {
+        /** @var \Drupal\migrate\Plugin\MigrationInterface $migration */
+        $migration = $this->getMigrationPluginManager()->createInstance($migrationId);
+        if ($migration) {
+            $status = $migration->getStatus();
+            if ($status == MigrationInterface::STATUS_IDLE) {
+                $this->logger()->warning(dt('Migration @id is already Idle', ['@id' => $migrationId]));
+            } else {
+                $migration->setStatus(MigrationInterface::STATUS_IDLE);
+                $this->logger()->log('status', dt('Migration @id reset to Idle', ['@id' => $migrationId]));
```

Comment:
We usually use the success() method here instead of log(). In general we prefer the specific methods to rather than calling log($LEVEL)